ta/subkey_notes.txt

```diff
@@ -0,0 +1,71 @@
+Subkey
+======
+
+A TA or a subkey can be signed with a subkey instead of the root key.  Here
+we're trying to describe the relationship between the different subkeys and
+keys used for the subkey test TAs.
+
+We start with a two level subkey hierachy to be able to test more when the
+subkey chain consist of more than a single subkey signed with the root key.
+
+The relationship of the different subkeys and TAs are as follows:
+root_key
+|-- identiy_subkey2
+|   `-- subkey2_ta
+`-- top_level_subkey
+    `-- mid_level_subkey
+        `-- subkey1_ta
+
+The keys themselves where generated using:
+
+openssl genrsa -out top_level_subkey.pem
+openssl genrsa -out mid_level_subkey.pem
+openssl genrsa -out identity_subkey2.pem
+
+The top level subkey is signed with the root key using (where the UUID is
+generated):
+
+../../optee_os/scripts/sign_encrypt.py sign-subkey \
+	--uuid f04fa996-148a-453c-b037-1dcfbad120a6 \
+	--key ../../optee_os/keys/default_ta.pem --in top_level_subkey.pem \
+	--out top_level_subkey.bin --max-depth 4 --name-size 64 \
+	--subkey-version 1
+
+The mid level subkey has to have a UUID in the namespace of the top level
+level subkey. The UUID is determined as
+UUIDv5(f04fa996-148a-453c-b037-1dcfbad120a6, "mid_level_subkey") =
```

Comment:
UUIDv5_SHA512().
Perhaps also add "where UUIDv5_SHA512() denotes the UUIDv5 scheme with SHA512 used instead of SHA1 and truncated to 20 bytes", it doesn't hurt to be precise ;-)